### ISSUE TITLE
Default formatter doesn't format `<svelte:options />` properly

### ISSUE THREAD

**Issue by AdrianGonz97** [sveltejs/language-tools#2238]
### Describe the bug

If `<svelte:options />` is below the script, `<svelte:options />` will be hoisted up to the top, _but_ `</script>` will be replaced for a `</svelte:options>`.

I first assumed that this was an issue with the prettier plugin, but running `pnpm format` doesn't appear to cause the problem. The issue only arises when I format with the default formatter that's part of the VSCode extension.

### Reproduction

https://github.com/AdrianGonz97/svelte-options-format-reproduction

### Expected behaviour

to not replace the `</script>` tag

### System Info

- OS: WSL2
- IDE: VSCode
- Svelte for VS Code `v108.1.0`

### Which package is the issue about?

Svelte for VS Code extension

### Additional Information, eg. Screenshots

![](https://i.imgur.com/QxVwWZ9.gif)

If a comment stands inbetween the `<svelte:options />` element and the `script` tag, the file formats properly:

![](https://i.imgur.com/OgDSdj5.gif)

**Comment by jasonlyu123** [comment]
I can't reproduce this. Can you check if it's still happening? And check if you have any vscode prettier config.

**Comment by huntabyte** [comment]
I'm able to reproduce this:

https://github.com/sveltejs/language-tools/assets/64506580/8b822c01-c1ab-4034-86d3-1b394a9c78e4

Prettier config:
```json
{
	"useTabs": true,
	"singleQuote": true,
	"trailingComma": "es5",
	"printWidth": 100,
	"plugins": ["prettier-plugin-svelte"],
	"overrides": [{ "files": "*.svelte", "options": { "parser": "svelte" } }]
}
```

```
"prettier": "^3.1.1",
"prettier-plugin-svelte": "^3.1.2"
```

**Comment by AdrianGonz97** [comment]
Thanks for taking a look!

> Can you check if it's still happening?

Unfortunately, it's still happening for me. I also performed an extension bisect to isolate the svelte vscode plugin and the issue still occurs. 

> And check if you have any vscode prettier config.

I only have one prettier specific setting, which is `prettier.useTabs`. However, here are all the svelte related settings I have set:
```jsonc
{
	// prettier specific
	"prettier.useTabs": true,

	// svelte vscode extension specific
	"[svelte]": {
		"editor.formatOnSave": true,
		"editor.defaultFormatter": "svelte.svelte-vscode"
	},
	"svelte.enable-ts-plugin": true,
	
	// svelte related misc
	"eslint.validate": [
		"svelte",
		"typescript",
		"javascript",
	],
	"explorer.fileNesting.patterns": {
		"+page.svelte": "+page.js, +page.ts, +page.server.js, +page.server.ts",
		"+layout.svelte": "+layout.js, +layout.ts, +layout.server.js, +layout.server.ts"
	}	
}
```

**Comment by jasonlyu123** [comment]
I still can't reproduce it. Do you have any config to run other actions when the file is saved? For example, `editor.codeActionsOnSave`? Also, does this happen in a clean setup? You can create an empty setting profile to test it so it's easier to switch back after testing.

**Comment by AdrianGonz97** [comment]
> Do you have any config to run other actions when the file is saved? For example, `editor.codeActionsOnSave`?

Seems like I do, however disabling it doesn't seem to fix it:
```jsonc
"editor.codeActionsOnSave": {
	"source.sortImports": "never"
},
```

> Also, does this happen in a clean setup? You can create an empty setting profile to test it so it's easier to switch back after testing.

Interesting! Didn't know this was a thing. I created a temporary profile, installed the Svelte extension and everything seems to be working as intended now 🤔. 

---

Aha! I found the trouble maker. Thanks for pointing me in the right direction. 

[Auto Rename Tag](https://marketplace.visualstudio.com/items?itemName=formulahendry.auto-rename-tag) seems to be the hooligan in this case. Disabling it makes everything work perfectly again.

Also, it seems like bisecting vscode extensions while in WSL2 doesn't disable the vscode extensions that are installed in `Local`, which explains why the issue persisted 🤦. 

Thanks again for your help @jasonlyu123!
